class-components/src/types/SearchTypes.ts

```diff
@@ -0,0 +1,46 @@
+import { ReactNode } from 'react';
+
+export interface ISearchProps {
+  onSearch: (query: string) => void;
+}
+
+export interface ISearchState {
+  query: string;
+}
+
+export interface ISingleResult {
```

Comment:
```suggestion
export interface ICharacter  {
```
Obviously it's a single item, bad naming